### ISSUE TITLE
Missing functional padding: expose no_slip_boundaries, no_flux_boundaries, zero_gradient_boundaries as public API

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#146]
Somax exposes easy-to-use wrappers for padding arrays to enforce boundary conditions, e.g.:

```python
# somax/_src/boundaries/__init__.py
pad_width = ((1, 1), (1, 1))

def zero_boundaries(u, pad_width):
    return jnp.pad(array=u, pad_width=pad_width, mode="constant", constant_values=0.0)

def zero_gradient_boundaries(u, pad_width):
    return jnp.pad(array=u, pad_width=pad_width, mode="edge")

def no_flux_boundaries(u, pad_width):
    return jnp.pad(array=u, pad_width=pad_width, mode="constant", constant_values=0.0)
```
Source: https://github.com/jejjohnson/somax/blob/8dc7229fa57dde0366986b97b6fe7372e3da21e0/somax/_src/boundaries/__init__.py

finitevolX provides primitive ghost-cell filling via boundary condition objects, but it lacks explicit functional wrappers for direct padding in an axis-controllable way (pad_width argument), to facilitate model code refactor from somax.

**Request:**
- Add public functional wrappers for `no_slip_boundaries`, `no_flux_boundaries`, and `zero_gradient_boundaries`, supporting arbitrary axes and pad_width control.
- Also mirror axis-directional and stagger-aware variants from somax:
  - [`boundaries/kinematic.py`](https://github.com/jejjohnson/somax/blob/8dc7229fa57dde0366986b97b6fe7372e3da21e0/somax/_src/boundaries/kinematic.py): `(u, axes)` wrappers (`free_slip_padding`, `no_slip_padding`) with `(left/right/both/None)` descriptors
  - [`models/sw/boundaries.py`](https://github.com/jejjohnson/somax/blob/8dc7229fa57dde0366986b97b6fe7372e3da21e0/somax/_src/models/sw/boundaries.py): sign-flipping true no-slip `no_slip_padding` (ghost = -interior)
- Please clarify the relationship between ghost-cell ring ([1:-1, 1:-1]) and explicit padding mode for interior arrays (Somax vs. finitevolX BC).
- Documentation should include guidance for physical BCs (true no-slip vs. zero-padding vs. zero-gradient, axis and staggerings).

**Open questions for future clarification:**
- Should the sign-flipping variant be included in the public API? (see `models/sw/boundaries.py`)
- Should axis-directional descriptors be unified across APIs for easier refactor?
- How should stagger-specific BC wrappers (for face and node locations) operate in the context of finitevolX stencil layout?

**Goal:**
Enable seamless refactor for Somax boundary calls to finitevolX primitives directly.

---

**References:**  
- Somax implementation: [`boundaries/__init__.py`](https://github.com/jejjohnson/somax/blob/8dc7229fa57dde0366986b97b6fe7372e3da21e0/somax/_src/boundaries/__init__.py) [`boundaries/kinematic.py`](https://github.com/jejjohnson/somax/blob/8dc7229fa57dde0366986b97b6fe7372e3da21e0/somax/_src/boundaries/kinematic.py) [`models/sw/boundaries.py`](https://github.com/jejjohnson/somax/blob/8dc7229fa57dde0366986b97b6fe7372e3da21e0/somax/_src/models/sw/boundaries.py)
- [Somax usage examples](https://github.com/jejjohnson/somax/search?q=zero_gradient_boundaries)
